### ISSUE TITLE
Some great feature requests.

### ISSUE THREAD

**Issue by TatsP** [MehdiRtal/MicrosoftRewardsBot#67]
I have a few feature ideas that have existed in bots previously, they are as follows:
1.) Multi-threading and parallel processing to run multiple accounts simultaneously.
2.) Auto account creation by getting random names and details from a site like https://namefake.com/ where only captcha needs to be solved manually or it can be solved automatically with the help of services like:
1.) http://www.imagetyperz.com/
2.) https://deathbycaptcha.com/
3.) https://2captcha.com/
and saving these details into a text file for later use.
3.) Auto-redeem rewards with URL or name for different countries along with phone verification using a service with available API such as:
1.) https://frenziedsms.com/
2.) https://otpzone.com/
3.) https://www.textverified.com/
If you could implement this, it would make the bot completely automated which would save a lot of time.

**Comment by DerMuffin** [comment]
Thanks for your input!

1.) In my opinion, Multi-Threading might be a difficult. Not to implement but to use. First thing is, most of the scripts run on low end hardware to keep the costs low. To run multiple accounts at once there is much more power needed or the scripts run much slower than befour. Second thing is, its kinda suspicious when all accounts get login from the same IP and do the same actions which lead in a much more quicker way to account ban.

2.) Auto Account creation is a great idea, manually create them takes a lot of time

3.) Auto redeem: Here i see the next problem. Unfortunately Microsoft detects this virtual Phone numbers. If one of your accounts is locked and you try to send a sms to one of theese numbers it just say "Cant use this Number to send SMS". To redeem a giftcard it might work at first, but i dont think you will ever see the giftcard you orderd because of Account suspension.

**Comment by Axel3026** [comment]
1) I manually connected to 5 profiles from the same IP and farmed 3 months ago and they didn't get banned, I did it semi-automatically.
2) something could be implemented to run multi-accounts, since before I thought that the --session argument was to open multi-accounts but later I realized that it was not like that

**Comment by Axel3026** [comment]
> 
How strange, but luckily and for the moment nothing happened to me since I farm in the USA and Germany.

**Comment by Axel3026** [comment]
Now I wanted to have this multi-account thing, but that you can choose how many accounts are farmed at the same time, if you want to farm 3 simultaneously or 5 or more. I'd love to. Since currently I have to open 5 Edge browsers to farm Germany and so on until I complete the 10 accounts.

**Comment by dmarts05** [comment]
> I have a few feature ideas that have existed in bots previously, they are as follows: 1.) Multi-threading and parallel processing to run multiple accounts simultaneously. 2.) Auto account creation by getting random names and details from a site like https://namefake.com/ where only captcha needs to be solved manually or it can be solved automatically with the help of services like: 1.) http://www.imagetyperz.com/ 2.) https://deathbycaptcha.com/ 3.) https://2captcha.com/ and saving these details into a text file for later use. 3.) Auto-redeem rewards with URL or name for different countries along with phone verification using a service with available API such as: 1.) https://frenziedsms.com/ 2.) https://otpzone.com/ 3.) https://www.textverified.com/ If you could implement this, it would make the bot completely automated which would save a lot of time.

This all seem great ideas, I have special interest in trying to implement the third one. It shouldn't be that challenging to get that last one working since they all have working APIs. Thank you for all your suggestions!

**Comment by mgrimace** [comment]
In terms of project suggestions, would a discord for ideas, suggestions, support be anything folks are interested in? I'm not sure if that level of availability is a good or bad thing, but might be interesting to swap farming strategies, ideas, etc. What do folks think?

**Comment by TatsP** [comment]
> In terms of project suggestions, would a discord for ideas, suggestions, support be anything folks are interested in? I'm not sure if that level of availability is a good or bad thing, but might be interesting to swap farming strategies, ideas, etc. What do folks think?

Yes I sure do think a Discord community for such strategies would be a great thing but it has to be moderated where only people that actually provide a valuable input should be allowed.

**Comment by TatsP** [comment]
> > I have a few feature ideas that have existed in bots previously, they are as follows: 1.) Multi-threading and parallel processing to run multiple accounts simultaneously. 2.) Auto account creation by getting random names and details from a site like https://namefake.com/ where only captcha needs to be solved manually or it can be solved automatically with the help of services like: 1.) http://www.imagetyperz.com/ 2.) https://deathbycaptcha.com/ 3.) https://2captcha.com/ and saving these details into a text file for later use. 3.) Auto-redeem rewards with URL or name for different countries along with phone verification using a service with available API such as: 1.) https://frenziedsms.com/ 2.) https://otpzone.com/ 3.) https://www.textverified.com/ If you could implement this, it would make the bot completely automated which would save a lot of time.
> 
> These all seem like great ideas, I have special interest in trying to implement the third one. It shouldn't be that challenging to get that last one working since they all have working APIs. Thank you for all your suggestions!
> 
> I will probably need some testers though.

I would be happy to be a tester.

**Comment by dmarts05** [comment]
> > > I have a few feature ideas that have existed in bots previously, they are as follows: 1.) Multi-threading and parallel processing to run multiple accounts simultaneously. 2.) Auto account creation by getting random names and details from a site like https://namefake.com/ where only captcha needs to be solved manually or it can be solved automatically with the help of services like: 1.) http://www.imagetyperz.com/ 2.) https://deathbycaptcha.com/ 3.) https://2captcha.com/ and saving these details into a text file for later use. 3.) Auto-redeem rewards with URL or name for different countries along with phone verification using a service with available API such as: 1.) https://frenziedsms.com/ 2.) https://otpzone.com/ 3.) https://www.textverified.com/ If you could implement this, it would make the bot completely automated which would save a lot of time.
> > 
> > 
> > These all seem like great ideas, I have special interest in trying to implement the third one. It shouldn't be that challenging to get that last one working since they all have working APIs. Thank you for all your suggestions!
> > I will probably need some testers though.
> 
> I would be happy to be a tester.

Nice, I'll let you know if I finally do this.

**Comment by TatsP** [comment]
> > > > I have a few feature ideas that have existed in bots previously, they are as follows: 1.) Multi-threading and parallel processing to run multiple accounts simultaneously. 2.) Auto account creation by getting random names and details from a site like https://namefake.com/ where only captcha needs to be solved manually or it can be solved automatically with the help of services like: 1.) http://www.imagetyperz.com/ 2.) https://deathbycaptcha.com/ 3.) https://2captcha.com/ and saving these details into a text file for later use. 3.) Auto-redeem rewards with URL or name for different countries along with phone verification using a service with available API such as: 1.) https://frenziedsms.com/ 2.) https://otpzone.com/ 3.) https://www.textverified.com/ If you could implement this, it would make the bot completely automated which would save a lot of time.
> > > 
> > > 
> > > These all seem like great ideas, I have special interest in trying to implement the third one. It shouldn't be that challenging to get that last one working since they all have working APIs. Thank you for all your suggestions!
> > > I will probably need some testers though.
> > 
> > 
> > I would be happy to be a tester.
> 
> Nice, I'll let you know if I finally do this.

There are 2 more issues I found in the bot:
1.) The bot does not click on all the daily activities, it skips activities such as Install Edge Browser Addon, Donation activities, etc.
These activities pay when you click on them without installing any browser and addons as I have manually confirmed it and this makes the bot lose 20-30 points.
2.) The bot does not click on the banners such as "Here are 100 points for reaching level 2" or "Here are 100 points for setting your goal". These make the bot lose a substantial amount of points whenever such banners are available.

**Comment by abhish2** [comment]
> > In terms of project suggestions, would a discord for ideas, suggestions, support be anything folks are interested in? I'm not sure if that level of availability is a good or bad thing, but might be interesting to swap farming strategies, ideas, etc. What do folks think?
> 
> Yes I sure do think a Discord community for such strategies would be a great thing but it has to be moderated where only people that actually provide a valuable input should be allowed.

how about we ourselves make one and moderate. People experienced in programming can imply ideas and suggestions in realtime rather than waiting for hours/days for reply on github.

**Comment by MehdiRtal** [comment]
Yes, we can start a discord server.
Contact me on discord: Sackuro#5802
